src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -98,99 +146,197 @@ internal abstract partial class AnalyzerDriver : IDisposable
+        {
+            get
+            {
+                Debug.Assert(_lazyIsGeneratedCodeSymbolMap != null);
+                return _lazyIsGeneratedCodeSymbolMap;
+            }
+        }
 
         /// <summary>
         /// Lazily populated dictionary indicating whether a source file has any hidden regions - we populate it lazily to avoid realizing all syntax trees in the compilation upfront.
         /// </summary>
-        private ConcurrentDictionary<SyntaxTree, bool> _lazyTreesWithHiddenRegionsMap;
+        private ConcurrentDictionary<SyntaxTree, bool>? _lazyTreesWithHiddenRegionsMap;
 
         /// <summary>
         /// Symbol for <see cref="System.CodeDom.Compiler.GeneratedCodeAttribute"/>.
         /// </summary>
-        private INamedTypeSymbol _generatedCodeAttribute;
+        private INamedTypeSymbol? _lazyGeneratedCodeAttribute;
 
         /// <summary>
         /// Driver task which initializes all analyzers.
         /// This task is initialized and executed only once at start of analysis.
         /// </summary>
-        private Task _initializeTask;
+        private Task? _lazyInitializeTask;
 
         /// <summary>
-        /// Flag to indicate if the <see cref="_initializeTask"/> was successfully started.
+        /// Flag to indicate if the <see cref="_lazyInitializeTask"/> was successfully started.
         /// </summary>
         private bool _initializeSucceeded = false;
 
         /// <summary>
         /// Primary driver task which processes all <see cref="CompilationEventQueue"/> events, runs analyzer actions and signals completion of <see cref="DiagnosticQueue"/> at the end.
         /// </summary>
-        private Task _primaryTask;
+        private Task? _lazyPrimaryTask;
 
         /// <summary>
         /// Number of worker tasks processing compilation events and executing analyzer actions.
         /// </summary>
         private readonly int _workerCount = Environment.ProcessorCount;
 
+        private AsyncQueue<CompilationEvent>? _lazyCompilationEventQueue;
+
         /// <summary>
         /// Events queue for analyzer execution.
         /// </summary>
-        public AsyncQueue<CompilationEvent> CompilationEventQueue { get; private set; }
+        public AsyncQueue<CompilationEvent> CompilationEventQueue
+        {
+            get
+            {
+                Debug.Assert(_lazyCompilationEventQueue != null);
+                return _lazyCompilationEventQueue;
+            }
+        }
+
+        private DiagnosticQueue? _lazyDiagnosticQueue;
 
         /// <summary>
         /// <see cref="DiagnosticQueue"/> that is fed the diagnostics as they are computed.
         /// </summary>
-        public DiagnosticQueue DiagnosticQueue { get; private set; }
+        public DiagnosticQueue DiagnosticQueue
+        {
+            get
+            {
+                Debug.Assert(_lazyDiagnosticQueue != null);
+                return _lazyDiagnosticQueue;
+            }
+        }
```

Comment:
💡 Rather than use an assertion here, allow the nullable flow analysis to fully operate by applying `[MemberNotNull(nameof(DiagnosticQueue))]` to both `Initialize` and a new `IsInitialized` property. This applies widely to the properties of this file.

Reply:
So all places that access these properties would need `Debug.Assert(IsInitialized)`? That is almost the entire file. I find this much cleaner and declarative - there is a lazily initialized field which is only intended to be written except for the assert here and a corresponding readonly property backed by it such that all accesses need to be on the initialized path.

Reply:
Also note that there are many such lazy fields in this type, and some of their initializations depend on values and order of other lazy initializations. A global `IsInitialized` will not help catch all potential bugs here.